ocaml/libs/xapi-stdext/lib/xapi-fdcaps/operations.ml

```diff
@@ -207,7 +207,7 @@ let with_tempfile ?size () f =
     try Unix.unlink name with Unix.Unix_error (_, _, _) -> ()
   in
   let@ () = Fun.protect ~finally in
-  let t = ch |> Unix.descr_of_out_channel |> make_wo_exn `reg in
+  let t = ch |> Unix.descr_of_out_channel |> Unix.dup |> make_wo_exn `reg in
   let@ t = with_fd t in
```

Comment:
The remaining is protected by the `finally` above already, why using `with_fd` again to protect the `t`?
How about to drop this line directly?

Reply:
the `finally` above is used to protect `ch` against leaks, whereas the `width_fd` here protects the newly created `t`. Which is not the same as `ch`, because we've just dup-ed it.